bootstrap/gamma/gamma_e2e_test.go

```diff
@@ -104,3 +104,28 @@ func TestGammaWithEmptyBlocks(t *testing.T) {
 	t.Run("Benchmark", testGammaWithEmptyBlocks(`{"transaction-pool-time-between-empty-blocks":"200ms"}`))
 	t.Run("LeanHelix", testGammaWithEmptyBlocks(fmt.Sprintf(`{"active-consensus-algo":%d,"transaction-pool-time-between-empty-blocks":"200ms"}`, consensus.CONSENSUS_ALGO_TYPE_LEAN_HELIX)))
 }
+
+func TestGammaSetBlockTime(t *testing.T) {
+	randomPort := test.RandomPort()
+	runMain(t, randomPort, "")
+	endpoint := fmt.Sprintf("http://0.0.0.0:%d", randomPort)
+
+	require.True(t, test.Eventually(WAIT_FOR_BLOCK_TIMEOUT, waitForBlock(endpoint, 1)))
+
+	res, err := http.Post(endpoint+"/debug/gamma/inc-time?seconds-to-add=10", "text/plain", nil)
+	require.NoError(t, err, "failed incrementing next block time")
+	require.EqualValues(t, 200, res.StatusCode, "http call to increment time failed")
+
+	sender, _ := orbsClient.CreateAccount()
+	transferTo, _ := orbsClient.CreateAccount()
+	client := orbsClient.NewClient(endpoint, 42, codec.NETWORK_TYPE_TEST_NET)
+
+	desiredTime := time.Now().Add(10 * time.Second)
```

Comment:
why 10 seconds? a major potential complication with this feature is clashing with logic in consensus algorithm which may rely on the system clock. if this feature is meant to be robust with lean-helix I'd run this test once for each consensus algorithm, and with clock deltas which are far larger then any config constant. for example, 10 minutes, or maybe even 45 minutes